python/lsst/afw/cameraGeom/camera.py

```diff
@@ -106,8 +129,38 @@ def findDetectors(self, cameraPoint):
                 detectorList.append(detector)
         return detectorList
 
+    def findDetectorsArray(self, xCoord, yCoord, coordSys):
+        """!Find the detectors that cover a list of points specified by x and y coordinates in any system
+
+        @param[in] xCoord is an array of x coordinates
+        @param[in] yCoord is an array of y coordinates
+        @param[in] coordSys is the coordinate system in which xCoord and yCoord are defined
+        @return a list of lists; each list contains the names of all detectors which contain the
+        corresponding point
+        """
+
+        cameraPoints = [afwGeom.Point2D(x,y) for x,y in zip(xCoord, yCoord)]
+
+        #transform the points to the native coordinate system
+        nativePoints = self._transformSingleSysArray(cameraPoints, coordSys, self._nativeCameraSys)
+
+        detectorList = []
+        for i in range(len(xCoord)):
+            detectorList.append([])
```

Comment:
Better to use list comprehension:
    detectorList = [[] for _ in range(len(xCoord))]

There's no check that xCoord and yCoord have the same length.